### PR TITLE
Allow configuration of plugins/cmds using METAFLOW_ENABLED_* variable

### PR DESCRIPTION
**Pull request by romain-intel**
Users can now set in their configuration or pass via an environment variable, the plugins they want to be enabled.

As an example METAFLOW_ENABLED_STEP_DECORATORS=batch,catch,retry would enable only those three step decorators and remove all others.

Replaces #1171

### CODE REVIEW

File: metaflow/cmd/__init__.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from metaflow.extension_support import _ext_debug
+
+
+def add_cmd_support(g, base_init=False):
+    # See the similar add_plugin_support function in plugin/__init__.py
```
Comment (romain-intel) [inline metaflow/cmd/__init__.py:5]: fix: plugins/__init__.py

File: metaflow/extension_support.py [hunk 2]
```diff
@@ -994,9 +997,9 @@ def __init__(self, handled):
     def find_spec(self, fullname, path, target=None):
         # If we are trying to load a shadowed module (ending in ._orig), we don't
         # say we handle it
-        _ext_debug(
-            "Looking for %s in %s with target %s" % (fullname, str(path), target)
-        )
+        # _ext_debug(
```
Comment (romain-intel) [inline metaflow/extension_support.py:1000]: removed as this was too verbose

File: metaflow/plugins/__init__.py [hunk 3]
```diff
@@ -14,174 +205,141 @@ def _merge_lists(base, overrides, attr):
...
-    "AWS_CLIENT_PROVIDERS": (
-        [],
-        lambda base, overrides: _merge_lists(base, overrides, "name"),
-    ),
-    "get_plugin_cli": (lambda l=None: [] if l is None else l(), _merge_funcs),
-}
+def _lazy_plugin_resolve(category):
+    # We look at TOGGLE_<category> in metaflow_config and build a list of plugins
+    # we need
+    import metaflow.metaflow_config as config
```
Comment (romain-intel) [inline metaflow/plugins/__init__.py:211]: fix: need to remove this line and update comment

**Review by romain-intel — COMMENTED** [review]
A few more cleanups to do. I'll wait for more comments before pushing an update.

**Review by savingoyal — APPROVED** [review]
lgtm! let's merge and release once our internal tests pass - roughly 2 hrs from now

**Pull request merged**